### PR TITLE
Create microsite

### PR DESCRIPTION
**Pull request by mtsokol**
#15 
@mijicd Hi, I've just completed first draft of mdoc microsite and I would be grateful for a review and advice.
While working on this part I have encountered few issues which I would like to ask about.

So the general design is based on `ZIO`'s microsite. The `Essentials` are similar to `Data Types` where basic API usage is shown so it's going to be about library design. 

For now there are introduction page where the NIO idea is repeated and some references,
page about `SocketChannel`s and page about `FileChannel`s where simple operations are shown.

The `Use Cases` show usage patterns in more complex cases.

The one will be the already existing example of `Client-Server`.

For the second example, as you suggested, I've used `zio-actor`. I've decided to implement `actor-remote` module, just like basic `akka-remote`.

There will maybe appear something more for each section.

---
###  Questions that arose

1. Is such microsite ok? As mentioned I've followed `ZIO` microsite

2. For `Essentials` which data types should be also explained? `AsynchronousChannelGroup` or `Pipe` or `Selector`?

3. With `Actor Remoting` I've encountered few issues - right now the example does not compile:
- First because the snippets from `zio-actor` test suite fail to compile (!) https://github.com/zio/zio-actors/issues/47 (I'm using latest maven artifact: `0.0.1+10-51dde802`)
- Second the latest maven `zio-actor` artifact is way behind master because it uses `scalaz.zio` instead of `zio.zio` and the `for` comprehensions fail to compile due to different monads used.

How to deal with it? Build the master on my own or wait until new release?

4. Also in `zio-nio` on `test` none is executed and neither with expilcite `test:runMain` command  

5. Once again about this `Actor Remoting` example. What do you think about attempt to initialize such module for `zio` ecosystem? like `zio-remote` similar to `akka-remote` where such functionality could be implemented? 
---

As I said it's WIP but it builds via `docs/docusaurusCreateSite` and `yarn start` after `cd /website`

### CODE REVIEW

**Comment by mijicd** [comment]
@mtsokol First of all, thanks for the contribution! I think you misunderstood my suggestion. When I mentioned `zio-actors` I meant their docs, not implementing the functionality (you're more than welcome to grab [this one](https://github.com/zio/zio-actors/issues/19)). Thus being said, starting the review :).

File: build.sbt [hunk 1]
```diff
@@ -36,3 +36,19 @@ lazy val zioNio = project
...
+lazy val docs = project
+  .in(file("zio-nio-docs"))
+  .settings(
+    skip.in(publish) := true,
+    moduleName := "zio-nio-docs",
+    scalacOptions -= "-Yno-imports",
+    scalacOptions -= "-Xfatal-warnings",
+    libraryDependencies ++= Seq(
+      "dev.zio"      %% "zio"        % "1.0.0-RC11-1",
+      "dev.zio"      %% "zio-actors" % "0.0.1+10-51dde802",
```
Comment (mijicd) [inline build.sbt]: As said in the comment, `zio-actors` were pointed for the sake of documentation, no need to implement remoting here.

File: docs/essentials/files.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import zio.console._
+```
+
+## Basic operations 
+
+Opening file for given path and no additional open attributes:
+
+```scala mdoc:silent
+val channelM = for {
+  path <- ZIO.succeedLazy(Paths.get("./file.txt"))
```
Comment (mijicd) [inline docs/essentials/files.md]: `succeedLazy` is about to be deprecated from `ZIO`, therefore I'd argue that we shouldn't include it in examples.

File: docs/essentials/files.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+```scala mdoc:silent
+for {
+  channel <- channelM
+  fileLock <- channel.lock
+  _ = fileLock.isShared                      //false
+  _ = fileLock.release()
+    
+  fileLock <- channel.lock(position = 0, size = 10, shared = false)
+  _ = fileLock.overlaps(5, 20)               //true
+  _ = fileLock.close()                       //same as release
```
Comment (mijicd) [inline docs/essentials/files.md]: Side note: we should manage resources via `ZManaged`. As far as I'm aware of, this issue hasn't been migrated from old repo yet.

@quelgar @pshemass ^

File: docs/essentials/index.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+---
+
+ZIO-NIO is a small, unopinionated ZIO interface to NIO. 
+
+ - **[File Channel](files.md)** — For processing files that are available locally. For every operation a new fiber is started to perform operation
+ - **[Socket Channel](sockets.md)** — Provides API for remote communication with `InetSocket`s 
+ 
+## So what is NIO once again? 
+
+As standard input-output processing we are used to stream-like operations.
```
Comment (mijicd) [inline docs/essentials/index.md]: This section sounds a bit unstructured. Personally, I'd even consider skipping it and just referencing the relevant Javadocs, or keeping it shorter.

File: docs/usecases/actor_remoting.md [hunk 5]
```diff
@@ -0,0 +1,204 @@
+---
+id: usecases_actor_remoting
+title:  "Actor Remoting"
```
Comment (mijicd) [inline docs/usecases/actor_remoting.md]: This is amazing effort, but as said, it was caused by misunderstanding. I propose that apply the idea [here](https://github.com/zio/zio-actors/issues/19), and to provide some simpler examples here.

File: build.sbt (hunk 1 above)
Comment (mtsokol) [reply build.sbt]: Sure, removed

File: docs/essentials/files.md (hunk 2 above)
Comment (mtsokol) [reply docs/essentials/files.md]: Changed to `effectTotal`

File: docs/essentials/files.md (hunk 3 above)
Comment (mtsokol) [reply docs/essentials/files.md]: Sure, as a resource it should be treated accordingly. 
Here I refactored it by using `bracket` (credits for @sinoz for showing it here #22) in the first part and `Managed` in the second part, for demonstration purposes.

File: docs/essentials/index.md (hunk 4 above)
Comment (mtsokol) [reply docs/essentials/index.md]: Sure, removed it and added nio javadocs link. It's not enough for clear comprehensive explanation and making it shorter won't help either.

File: docs/usecases/actor_remoting.md (hunk 5 above)
Comment (mtsokol) [reply docs/usecases/actor_remoting.md]: I've misunderstood it but I was also thinking about writing such module for `zio-actors`. I will take this piece to present this idea in issue or gitter.

But still I will provide some other examples there.

**Comment by mtsokol** [comment]
@mijicd I've addressed comments, also added that circleci mdoc job (with scaladocs generation). 
Going back to my `2.` question, which data types are also worth describing in `Essentials`?

File: docs/usecases/actor_remoting.md (hunk 5 above)
Comment (mijicd) [reply docs/usecases/actor_remoting.md]: It would be very useful, and there's already an issue, i've referenced it in my comment, so take a look!

**Comment by quelgar** [comment]
This is great! Your "Essentials" covers files and sockets, which I think are indeed the most important aspects. Good to go IMHO.

@mijicd great point about using `ZManaged`. Really we should be aiming to not have any `close` methods, correct?

**Comment by mijicd** [comment]
@quelgar That would be ideal. I'm willing to take that one on me, as soon as we have an open ticket :)

File: website/core/Footer.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+                  data-count-aria-label="# stargazers on GitHub"
+                  aria-label="Star this project on GitHub">
+                Star
+              </a>
+            </div>
+            <div>
+              <h5>Additional resources</h5>
+              <a
+                  href="https://javadoc.io/doc/dev.zio/zio_2.12/">
+                Scaladoc of zio
```
Comment (jczuchnowski) [inline website/core/Footer.js]: make uppercase ZIO

File: docs/usecases/client_server.md [hunk 7]
```diff
@@ -0,0 +1,6 @@
+---
+id: usecases_client_server
+title:  "Client Server"
+---
+
+TODO remake existing `channel.md`
```
Comment (jczuchnowski) [inline docs/usecases/client_server.md]: So what do we need here?

File: website/core/Footer.js (hunk 6 above)
Comment (mtsokol) [reply website/core/Footer.js]: Sure, corrected it

File: docs/usecases/client_server.md (hunk 7 above)
Comment (mtsokol) [reply docs/usecases/client_server.md]: I've moved here the existing `tut/channel.md`, split it into functions, added comment for each part and provided example output. Also I've removed the old one.

**Comment by jczuchnowski** [comment]
Looks good. I think we should merge it as fast as possible to enable more contributions to the documentation.
